frappe/database/sequence.py

```diff
@@ -0,0 +1,73 @@
+from frappe import db, conf, scrub
+
+
+def create_sequence(
```

Comment:
MariaDB has [Auto Increment](https://mariadb.com/kb/en/auto_increment/) and Postgres has [Serial](https://www.postgresql.org/docs/14/datatype-numeric.html#DATATYPE-SERIAL) 
they are one word datatypes/constraints.

```
CREATE TABLE animals (
     id MEDIUMINT NOT NULL AUTO_INCREMENT,
     name CHAR(30) NOT NULL,
     PRIMARY KEY (id)
 );
```
```
CREATE TABLE tablename (
    colname SERIAL
);
```

Okay, so Postgres uses sequences under the hood of serial. But it actually will do all tasks like mapping the right sequences calling next_val on correctly.

[auto-increment is persistent for MariaDB now](https://mariadb.com/kb/en/how-to-retain-auto-increment-value-on-server-restart-innodb/)

The part that will affect us is that auto-increment will be done [only on insert](https://mariadb.com/kb/en/how-to-retain-auto-increment-value-on-server-restart-innodb/), We can defer name references for DocTypes until then IMO.